number_parser/parser.py

```diff
@@ -298,14 +322,37 @@ def parse_fraction(input_string, language=None):
     return None
 
 
-def parse(input_string, language=None):
+def parse(input_string, language=None, numeral_systems=None):
     """
     Converts all the numbers in a sentence written in natural language to their numeric type while keeping
     the other words unchanged. Returns the transformed string.
     """
+    complete_sentence = None
+
+    if numeral_systems is None:
+        numeral_systems = NUMERAL_SYSTEMS
+
     if language is None:
         language = _valid_tokens_by_language(input_string)
 
+    temporary_sentence = input_string
+    for numeral_system in numeral_systems:
+
+        if numeral_system == 'decimal':
+            complete_sentence = _parse_decimal(temporary_sentence, language)
+            temporary_sentence = complete_sentence
+
+        elif numeral_system == 'roman':
+            complete_sentence = _parse_roman(temporary_sentence)
+            temporary_sentence = complete_sentence
+
+        else:
+            raise ValueError(f'"{numeral_system}" is not a supported numeral system')
+
+    return complete_sentence
```

Comment:
The new names still make you use 2 lines for something that could be expressed in 1 and remove 1 variable. In this case I think the simplification improves readability slightly.

```suggestion
    result = input_string
    for numeral_system in numeral_systems:

        if numeral_system == 'decimal':
            result = _parse_decimal(result, language)

        elif numeral_system == 'roman':
            result = _parse_roman(result)

        else:
            raise ValueError(f'"{numeral_system}" is not a supported numeral system')

    return result
```